compiler/IREmitter/Payload/patches/vm_insnhelper.c

```diff
@@ -978,6 +979,24 @@ VALUE sorbet_vm_instance_variable_get(struct FunctionInlineCache *getCache, IVC
     return sorbet_callFuncWithCache(getCache, VM_BLOCK_HANDLER_NONE);
 }
 
+VALUE sorbet_vm_instance_variable_set(struct FunctionInlineCache *setCache, IVC varCache, rb_control_frame_t *reg_cfp,
+                                      VALUE recv, ID var, VALUE value) {
+    sorbet_vmMethodSearch(setCache, recv);
+    rb_method_definition_t *setDef = setCache->cd.cc.me->def;
+    if (setDef->type == VM_METHOD_TYPE_CFUNC &&
+        setDef->body.cfunc.func == sorbet_vm_Kernel_instance_variable_set_func()) {
+        sorbet_vm_setivar(recv, var, value, varCache);
+        return value;
+    }
+
+    VALUE *sp = reg_cfp->sp;
+    *(sp + 0) = recv;
+    *(sp + 1) = rb_id2sym(var);
+    *(sp + 2) = value;
```

Comment:
Might be better to use `vm_env_write` here, as it'll be available in this context.

Reply:
Isn't `vm_env_write` intened to be used for writing local variables for the current stack frame?  These are intended to be arguments for the call, not local variables.

Reply:
Good point. I remembered there being a helper for pushing values to the ruby stack, but looking at `vm_call0_body` it looks like such a thing doesn't exist.